pkg/helm/helm3lib/helm3lib.go

```diff
@@ -0,0 +1,315 @@
+	// No need to delete single failed revision anymore
+	// https://github.com/helm/helm/issues/8037#issuecomment-622217632
+	return nil
+}
+
+func (h *LibClient) DeleteOldFailedRevisions(releaseName string) error {
+	// No need to delete single failed revision anymore
+	// https://github.com/helm/helm/issues/8037#issuecomment-622217632
+	return nil
+}
+
+// LastReleaseStatus returns last known revision for release and its status
+func (h *LibClient) LastReleaseStatus(releaseName string) (revision string, status string, err error) {
+	release, err := actionConfig.Releases.Last(releaseName)
+	if err != nil {
+		if err == driver.ErrReleaseNotFound || strings.HasPrefix(err.Error(), "no revision for release") {
+			return "0", "", fmt.Errorf("release '%s' not found\n", releaseName)
+		}
+		return "", "", err
+	}
+
+	return strconv.FormatInt(int64(release.Version), 10), release.Info.Status.String(), nil
+}
+
+func (h *LibClient) UpgradeRelease(releaseName string, chartName string, valuesPaths []string, setValues []string, namespace string) error {
+	upg := action.NewUpgrade(actionConfig)
+	if namespace != "" {
+		upg.Namespace = namespace
+	}
+
+	upg.Install = true
+	upg.MaxHistory = int(options.HistoryMax)
+	upg.Timeout = options.Timeout
+
+	chart, err := loader.Load(chartName)
+	if err != nil {
+		return err
+	}
+
+	var resultValues chartutil.Values
+
+	for _, vp := range valuesPaths {
+		values, err := chartutil.ReadValuesFile(vp)
+		if err != nil {
+			return err
+		}
+
+		resultValues = chartutil.CoalesceTables(resultValues, values)
+	}
+
+	if len(setValues) > 0 {
+		m := make(map[string]interface{})
+		for _, sv := range setValues {
+			arr := strings.Split(sv, "=")
+			if len(arr) == 2 {
+				m[arr[0]] = arr[1]
+			}
+		}
+		resultValues = chartutil.CoalesceTables(resultValues, m)
+	}
+
+	h.LogEntry.Infof("Running helm upgrade for release '%s' with chart '%s' in namespace '%s' ...", releaseName, chartName, namespace)
+	histClient := action.NewHistory(actionConfig)
+	histClient.Max = 1
+	lr, err := histClient.Run(releaseName)
+	if err == driver.ErrReleaseNotFound {
+		instClient := action.NewInstall(actionConfig)
+		if namespace != "" {
+			instClient.Namespace = namespace
+		}
+		instClient.Timeout = options.Timeout
+		instClient.ReleaseName = releaseName
+		instClient.UseReleaseName = true
+
+		_, err = instClient.Run(chart, resultValues)
+
+		return err
+	}
+	if len(lr) > 0 {
+		if lr[0].Info.Status.IsPending() {
+			rb := action.NewRollback(actionConfig)
+			_ = rb.Run(lr[0].Name)
+		}
+	}
+
+	_, err = upg.Run(releaseName, chart, resultValues)
+	if err != nil {
+		return fmt.Errorf("helm upgrade failed: %s\n", err)
+	}
+	h.LogEntry.Infof("Helm upgrade for release '%s' with chart '%s' in namespace '%s' successful", releaseName, chartName, namespace)
+
+	return nil
+}
+
+func (h *LibClient) GetReleaseValues(releaseName string) (utils.Values, error) {
+	gv := action.NewGetValues(actionConfig)
+	return gv.Run(releaseName)
```

Comment:
helm3 package set Namespace for 'get release values' operation. I think ns is required here.

Reply:
Same as for Uninstall... no ns here in attributes